### ISSUE TITLE
Feature request: Let Qiqqa suggest tags that are already defined by the user for other files

### ISSUE THREAD

**Issue by raindropsfromsky** [jimmejardine/qiqqa-open-source#183]
By definition, the user is going to use a different library for each distinct subject. 
In other words, all the files in a given library are supposed to be on a common subject.
Logically, these files will have common tags.

So, when I am browsing a new file, and if I want to attach tags to it, most of the tags which I have already attached to the other files in the current library would be applicable here as well.

But Qiqqa does not show me the list of existing tags to choose from.
I am forced to re-enter all tags afresh.

**Desired:** 
Show all the existing tags. 
Let the user select multiple tags and select **OK** to apply them all to the new file.

**Comment by GerHobbelt** [comment]
> By definition, the user is going to use a different library for each distinct subject.
> In other words, all the files in a given library are supposed to be on a common subject.

Not necessarily: for me, at least one library carries 20+ years of collected information on various topics, though my interests there are mostly focused around CS (computer science), Electronics and some Psychology/Medical research. Anyway, in that sense, I have at least one library that's very much akin to the physical analogue in the real world. 😉   I consider it my "local google pile", i.e. I search in my library when I "know" or otherwise *expect* to find in there what I'm looking for. This is particularly useful to me because there's quite a few occasions where Google et al don't deliver while my local library *can* deliver on the same search criteria, either due to "overly broad" search terms or because I also have several pieces of material in there which are not available on the Net (or at least accessible to *me*: some stuff was found while on a tangent search, pure dumb luck one day or even scans of old documents I own or had otherwise access to in the past.

On the other hand, I also have a few Qiqqa libraries which cover a single topic -- got to say though that I always have followed the practice of making sure everything that sits in those is also included in my single "master" library. Some of these "one subject" libraries have been constructed solely from material dug up from the master lib via search and tag selection, while others grew as a mix of actions: both picking material from the master library and external sources.

Anyway, a library is what you make of it. For me, a qiqqa library is not a one-purpose animal. Not per se.


> Logically, these files will have common tags.

Indeed,  many of the docs will share tags. IIRC, Qiqqa keeps track and suggests tags as you type. So I'm a bit confused whether this is a regression in the latest experimental release or some such. 🤔 

Haven't taken the time to check Qiqqa now. Will do that later.

*Or* this is a massive misunderstanding on my part about what this issue is about.

**Comment by raindropsfromsky** [comment]
Well, I too have a collection of 20+ years.

I have created a vast directory to reflect the ontology of the subjects. For example, the **Hobbies** folder gets split in multiple subfolders, such as **Photography**, **Drawing**, **Music**, etc. These in turn get split in more subfolders.  

And then I have moved all files to those subfolders.

Even then each folder has hundreds of files, and so if I try to tag all files in any subfolder, they won't be 100% same tags for all files. But there will be overlap.

BTW it is not sufficient to provide a "type-ahead" feature for tags. (When we type a few characters, Qiqqa will suggest the remaining entry). The reason for this is simple: We need to remember the entire superset of tags for each library. This is too much of a hassle. So Qiqqa should simplify the task by presenting the superset of tags, and let the user select the tags he wants. He can add new
 tags if needed.

**Comment by raindropsfromsky** [comment]
BTW I was experimenting with the GUI, and just noticed this: If we select multiple files in the library, the right-hand side panel changes to this:

![image](https://user-images.githubusercontent.com/9047168/77719977-9df1a800-700c-11ea-9f62-060c44c8359a.png)

Here, the intent is to let the user add and remove tags from all selected files.
But here also, the GUI does not provide any clues to the user as to which one he can remove.
He has to come here with premeditated intent.

There is no reason why he should not be allowed to see the superset of tags for the library.
Then he can take a decision comfortably.
